platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ExportMigrationEntry.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+
+/**
+ * The <code>ExportMigrationEntry</code> interfaces provides support for artifacts that are being
+ * exported during migration.
+ * <p>
+ * <b>
+ * This code is experimental. While this interface is functional
+ * and tested, it may change or be removed in a future version of the
+ * library.
+ * </b>
+ * </p>
+ */
+public interface ExportMigrationEntry extends MigrationEntry {
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified property
+     * in the properties file associated with this migration entry to be exported by the corresponding
+     * migratable.
+     * <p>
+     * An error will be automatically recorded with the associated migration report if the property
+     * is not defined in the property file referenced by this entry or its value is blank.
+     *
+     * @param name the name of the property referencing a migration entry to create or retrieve
+     * @return a new migration entry or the existing one if already created for the migration entry
+     * referenced from the specified property value or empty if the property is not defined or
+     * its value is blank
+     * @throws IllegalArgumentException if <code>name</code> is <code>null</code>
+     */
+    public default Optional<ExportMigrationEntry> getPropertyReferencedEntry(String name) {
+        return getPropertyReferencedEntry(name, (r, n) -> true);
+    }
+
+    /**
+     * Creates or retrieves (if already created) a migration entry referenced from the specified property
+     * in the properties file associated with this migration entry to be exported by the corresponding
+     * migratable.
+     * <p>
+     * The provided predicate will be invoked with the associated migration report and the property
+     * value (may be <code>null</code> if not defined). In case the property is not defined or is
```

Comment:
The predicate is the parameter passed to the getPropertyReferencedEntry(). What can be null is the value passed to the predicate. I updated the doc.